games/static/games/proto.js

```diff
@@ -204,18 +199,32 @@ class Board {
         }
     }
 
+    determineDirection() {
+        const aliensAtLeftBorder = this.squares[this.alienLocations[0]].isLeftBorder()
+        const aliensAtRightBorder = this.squares[this.alienLocations[this.alienLocations.length - 1]].isRightBorder()
+        const Left = -1
+        const Right = 1
+        if ((aliensAtLeftBorder && this.direction === Left) || (aliensAtRightBorder && this.direction === Right)) {
```

Comment:
please explain meanings
there are a lot of secret code here
what does `(aliensAtLeftBorder && this.direction === Left)` or `this.direction = this.width` means?
a lot of times interducing new one liner meaningful methods are super valuable for code readability and maintainability.
i prefare methods over variables